src/main/java/io/jenkins/plugins/github/checks/api/ChecksDetails.java

```diff
@@ -118,12 +165,29 @@ public ChecksDetailsBuilder withConclusion(final ChecksConclusion conclusion) {
             return this;
         }
 
-        public ChecksDetailsBuilder withOutputs(final List<Output> outputs) {
-            Objects.requireNonNull(outputs);
-            this.outputs = Collections.unmodifiableList(outputs);
+        /**
+         * Set the output of a check.
+         *
+         * @param output
+         *         an output of a check
+         * @return this builder
+         * @throws NullPointerException if the <code>outputs</code> is null
+         */
+        public ChecksDetailsBuilder withOutput(final Output output) {
+            // TODO: Should store the clone of the output after output is constructed.
+            Objects.requireNonNull(output);
```

Comment:
often it's nice to statically import requireNonNull so that the code read nicer
```suggestion
           requireNonNull(output);
```